File: packages/java/src/main/java/dev/kreuzberg/types/metadata/LinkMetadata.java
package dev.kreuzberg.types.metadata;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import java.util.Map;
import org.jspecify.annotations.Nullable;

/**
 * HTML link metadata.
 */
public record LinkMetadata(
    @JsonProperty("href") String href,
    @JsonProperty("text") String text,
    @JsonProperty("title") @Nullable String title,
    @JsonProperty("link_type") String linkType,
    @JsonProperty("rel") List<String> rel,
    @JsonProperty("attributes") Map<String, String> attributes
) { }

File: packages/php/src/Types/Metadata/LinkMetadata.php
<?php

declare(strict_types=1);

namespace Kreuzberg\Types\Metadata;

class LinkMetadata
{
    public string $href;
    public string $text;
    public ?string $title = null;
    public string $linkType = 'other';
    /** @var string[] */
    public array $rel = [];
    /** @var array<string, string> */
    public array $attributes = [];

    /**
     * @param string[] $rel
     * @param array<string, string> $attributes
     */
    public function __construct(
        string $href,
        string $text,
        string $linkType = 'other',
        ?string $title = null,
        array $rel = [],
        array $attributes = [],
    ) {
        $this->href = $href;
        $this->text = $text;
        $this->linkType = $linkType;
        $this->title = $title;
        $this->rel = $rel;
        $this->attributes = $attributes;
    }
}
